android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/surveyquestions/SurveyQuestionsViewModel.kt

```diff
@@ -54,8 +66,58 @@ class SurveyQuestionsViewModel(
     }
 
     fun answerQuestion(questionAndAnswers: QuestionAndAnswers) {
-        Timber.d(
-            "${questionAndAnswers.questionIndex} - ${questionAndAnswers.answerInputs}"
-        )
+        val uiModels = _questionContentUiModels.value.toMutableList()
+        val questionIndex = questionAndAnswers.questionIndex
+        val changedUiModel = uiModels[questionIndex]
+        changedUiModel.displayType.input = questionAndAnswers.answerInputs
+        uiModels[questionIndex] = changedUiModel
+        _questionContentUiModels.value = uiModels
+
+        // Convert answers to SurveySubmission
+        _survey?.let {
+            val questionSubmissions = questionContentUiModels.value
+                .mapIndexed { questionIndex, uiModel ->
+                    val questionId = questionId(questionIndex)
+                    val answerSubmissions: List<AnswerSubmission> =
+                        uiModel.displayType.input.map { input ->
+                            when (input) {
+                                is AnswerInput.Select -> AnswerSubmission(id = input.id)
+                                is AnswerInput.Content -> AnswerSubmission(
+                                    id = input.id,
+                                    answer = input.content
+                                )
+                            }
+                        }
+                    QuestionSubmission(
+                        questionId,
+                        answerSubmissions
+                    )
```

Comment:
then we just need to get it out from ui model

```suggestion
                    QuestionSubmission(
                        uiModel.id,
                        answerSubmissions
                    )
```